pkg/gardener/shoot/extender/provider.go

```diff
@@ -104,13 +105,19 @@ func NewProviderExtenderPatchOperation(enableIMDSv2 bool, defMachineImgName, def
 	}
 }
 
-func checkWorkerZonesMatchProviderConfig(providerType string, workerZones []string, ctrlPlaneConfig *runtime.RawExtension, infraConfig *runtime.RawExtension) error {
+func checkWorkerZonesMatchProviderConfig(providerType string, workerZones []string, ctrlPlaneConfig *runtime.RawExtension, infraConfig *runtime.RawExtension, patchValidation bool) error {
```

Comment:
Just one opinionated preference which is not a strong suggestion:

Have you considered adding one more function, such as `checkWorkerZonesMatchProviderConfigForPatch`, instead of adding a fifth parameter? However, that's also not a perfect option, as it involves some code duplication.

Reply:
Alternatively, if many parameters are needed but method definition gets too big, you can always do a small refactoring and switch to 

[ParamterObject](https://refactoring.guru/introduce-parameter-object)

Reply:
I can do it, but it will require to add additional 35 code lines only for this second function. So instead of adding 2 lines with simple fix - We are adding 37 lines and another function with very long name. Is it resonable?